pkg/segment/reader/segread/segstatsreader.go

```diff
@@ -190,216 +189,107 @@ func readSingleSst(fdata []byte, qid uint64) (*structs.SegStats, error) {
 func GetSegMin(runningSegStat *structs.SegStats,
 	currSegStat *structs.SegStats) (*utils.CValueEnclosure, error) {
 
-	result := utils.CValueEnclosure{}
-
 	if currSegStat == nil {
-		log.Errorf("GetSegMin: currSegStat is nil")
-		return &result, errors.New("GetSegMin: currSegStat is nil")
+		return &utils.CValueEnclosure{}, fmt.Errorf("GetSegMin: currSegStat is nil")
 	}
 
 	// if this is the first segment, then running will be nil, and we return the first seg's stats
 	if runningSegStat == nil {
-
-		if currSegStat.IsNumeric {
-			return currSegStat.NumStats.Min.ToCValueEnclosure()
-		} else {
-			if currSegStat.StringStats.Min.Dtype == utils.SS_DT_STRING {
-				return &utils.CValueEnclosure{
-					Dtype: utils.SS_DT_STRING,
-					CVal:  currSegStat.StringStats.Min.CVal,
-				}, nil
-			}
-		}
-
-		return &result, nil
+		return &currSegStat.Min, nil
 	}
 
-	// Prioritize numeric stats over strings
-	if !currSegStat.IsNumeric {
-		if runningSegStat.IsNumeric {
-			return runningSegStat.NumStats.Min.ToCValueEnclosure()
-		}
-
-		runningSegStat.StringStats.MergeMinStrStats(currSegStat.StringStats)
-
-		if runningSegStat.StringStats.Min.Dtype == utils.SS_DT_STRING {
-			result.Dtype = utils.SS_DT_STRING
-			result.CVal = runningSegStat.StringStats.Min.CVal
-		}
-
-		return &result, nil
+	result, err := utils.ReduceMinMax(runningSegStat.Min, currSegStat.Min, true)
+	if err != nil {
+		return &utils.CValueEnclosure{}, fmt.Errorf("GetSegMin: error in ReduceMinMax, err: %v", err)
 	}
-
-	if !runningSegStat.IsNumeric {
-		runningSegStat.NumStats = currSegStat.NumStats
+	runningSegStat.Min = result
+	if !runningSegStat.IsNumeric && runningSegStat.Min.IsNumeric() {
```

Comment:
Why would `IsNumeric` be false and `Min.IsNumeric` be true?

Reply:
The case where currSegStat is numeric but runnningSegStat is not. After we update the min/max we should update isNumeric parameter too.